### DISCUSSION TITLE
Wiki on how to style tooltips?

### DISCUSSION THREAD

**Discussion by ghost** [Alexays/Waybar#1824] [Q&A]
Tooltips are possible to customize, but they can't be inspected with a debug tool.
Example:
```
tooltip {
  color: black;
  background-color: white;
  text-shadow: none;
}

tooltip * {
  color: black;
  text-shadow: none;
}
```
But there is no way to inspect the elements to see what is inside those tags, so it's a guessing game for now.
For example, how do i change "today-format" element style in the "clock" module?

**Comment by poperigby** [comment]
I tried putting those snippets in my `style.css` but it didn't seem to change anything.

Edit: Nevermind, you're not supposed to have a `#` in front of `tooltip`.

**Comment by tasmo** [comment]
@PopeRigby Does it really work? On a quick code search I cannot find any input for style attributes for tooltips.

It would be great to have the possibility to stylize tooltips at all.

**Comment by poperigby** [comment]
Here's a CSS snippet:
```
tooltip {
     border-radius: 4px;
     background-color: #bd93f9;
}
```

Does that work?

**Comment by tasmo** [comment]
Thank you, almost. The `color:` property unfortunately does not.

**Comment by poperigby** [comment]
Sorry about that. Background color should be a hexadecimal, rgba, etc. value. I accidentally put in one of my defined colors.

**Comment by tasmo** [comment]
`background-color` works well even with a value coming from the GDK definition.

Just the foreground `color` does not work with any value not even a hexadecimal.

**Comment by poperigby** [comment]
Oh, then I misunderstood what you were saying.

**Comment by lokesh-krishna** [comment]
It's interesting how some properties work but some simple don't (font-family, font-size). What's the reason behind this behaviour?

**Comment by tasmo** [comment]
I found the solution for styling the foreground color:

```css
tooltip label {
  color: @theme_fg_color;
}
```

Added to the Wiki in https://github.com/Alexays/Waybar/wiki/Styling/9c4f6a764f5469d08d6a20695121156ee93dd3e5.

**Comment by lokesh-krishna** [comment]
How can one get padding to work for tooltips? Adding it as follows doesn't work. 
```CSS
tooltip {
  padding: 5px;
}

**Reply by shasherazi** [reply]
Did you find an answer?

**Reply by julihermes** [reply]
apply the padding on label, like the color solution above

**Comment by bd-g** [comment]
I wonder how big a refactor it would be to allow for tooltip config per module - for example, I want monospace font on some tooltips but not for others.

**Reply by ghost** [reply]
> I wonder how big a refactor it would be to allow for tooltip config per module - for example, I want monospace font on some tooltips but not for others.

Is it a way to style tooltips on a per-module basis?

**Comment by FragCGI** [comment]
You can setup custom CSS for tooltip by doing this
tooltip {
    background-color: transparent;
    border: none;
}

tooltip label {
    background-color: #color;
    color: #color;
    border-radius: ##px;
    border: style ##px #color;
    margin: ##px;
}

tooltip itself, for whatever reason, doesn't really accept any CSS settings, but tooltip label is the place to customize.
I haven't figured out how to do this for individual modules though.

**Reply by andeen171** [reply]
Nice, just the border-radius that doesn't seem to work

**Reply by RpxdYTX** [reply]
figured how to style the tooltip background:
```css
tooltip box.horizontal {
    border-radius: 10px;
}
```

**Comment by nathanchere** [comment]
Don't forget that CSS specificity still applies here. If you're trying to style something and it seems to be ignoring some of your rules but not others, since there's no `!important` the next best thing is something like this:

```
window#waybar #tray menu {
  border: 3px solid @border;
 }
```

**Comment by Benschar1** [comment]
You can style tooltips on a per-module basis by using pango markup in the tooltip text. So for a module like this:
```
"custom/module": {
  "format": ...,
  "exec": "path/to/script",
  "tooltip": true,
  "return-type": "json"
}
```

and a bash script like this:
```
text=...
formatted_tooltip="<tt><span color=\"yellow\">${tooltip text...}</span></tt>"

printf '{\"text\":\"%s\", \"tooltip\":\"%s\"}' "$text" "$formatted_tooltip" \
    | sed 's/$/\\n/g' \
    | tr -d '\n'

```

you'll get a yellow, monospaced tooltip for this module, and default tooltip styling for other modules. Just be careful with bash escapes when interpolating strings to create the json object.

**Comment by faervan** [comment]
Is it possible to add images to a tooltip?

**Comment by tmccombs** [comment]
Is it possible to target the tooltip for a specific module. 

For example, to use a monospace font for the calendar tooltip for the clock module?

**Reply by AyushmanOfficial** [reply]
> use a monospace font for the calendar tooltip for the clock module?

To do that for now you can do  
```css
"tooltip-format": "<tt><small><span size='12pt' font='GeistMono Nerd Font'>{calendar}</span></small></tt>"

**Comment by shasherazi** [comment]
Any idea on how to do padding and margin with tooltips?

**Reply by martincardenasdev** [reply]
Use the box selector,

```
tooltip box {
  padding: 12px;
}
```

**Reply by shasherazi** [reply]
Margins still dont work.

![Screenshot-2025-05-27_18:07:58](https://github.com/user-attachments/assets/6fdc31b8-05a1-4a6f-92f4-15a4b7a55ae6)

```css
tooltip box {
  background-color: #aaa;
  margin-right: 100px;
}
```

**Comment by kushkushal439** [comment]
Is there a way to change the activation time for tooltip, I want it to appear instantaneously on hovering.

**Comment by Sat-Nav** [comment]
If you adjust the tooltip too high (with negative margin-top), eventually it starts getting cut off, is there any way around this?

**Comment by bgmeulem** [comment]
Also having a lot of issues "guessing" tooltip targets. A wiki that shows the hierarchy would be of immense help

**Comment by dAeb099** [comment]
Guys, i don't have padding and border radius on tooltip despite I already wrote this code:

```css
tooltip {
  background-color: rgba(30, 30, 46, 0.85);
  border: 1px solid rgba(255, 255, 255, 0.2);
}
tooltip label {
  padding: 1px;
  color: #cdd6f4;
  border-radius: 10px;
}
```

What am I supposed to do? (`all: unset;` doesn't help)
